packages/cbioportal-frontend-commons/src/components/checkedSelect/Checklist.tsx

```diff
@@ -103,22 +145,182 @@ export default class Checklist extends React.Component<ChecklistProps, {}> {
             return;
         }
 
-        // only select current value if all selected already
-        if (
-            this.props.unselectOthersWhenAllSelected &&
-            this.props.value.length === this.props.options.length
+        // splice protein type badge was selected
+        if (value === 'splice_site') {
+            if (
+                this.props.unselectOthersWhenAllSelected &&
+                this.props.value.length === this.props.options.length
+            ) {
+                this.props.onChange([
```

Comment:
see how well we can compress these three cases.  ideally we could call ```this.props.onChange``` only once.